test/new-e2e/pkg/utils/e2e/client/agent_cmd_runner.go

```diff
@@ -37,12 +37,23 @@ func newAgentCommandRunner(t *testing.T, executeAgentCmdWithError executeAgentCm
 }
 
 func (agent *AgentCommandRunner) executeCommand(command string, commandArgs ...AgentArgsOption) string {
+
+	output, err := agent.executeCommandWithError(command, commandArgs...)
+	require.NoError(agent.t, err)
+	return output
+}
+
+func (agent *AgentCommandRunner) executeCommandWithError(command string, commandArgs ...AgentArgsOption) (string, error) {
+	if !agent.isReady {
+		err := agent.waitForReadyTimeout(1 * time.Minute)
+		require.NoErrorf(agent.t, err, "the agent is not ready")
+		agent.isReady = true
+	}
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
This part can be removed, we now wait for the agent to be ready at agent client init time

Reply:
Ok, just removed it

Reply:
In the end, I think we should keep it, otherwise it breaks when creating an AgentCommandRunner outside of the Agent env

Reply:
You could add the wait for ready option to `NewAgentCommandRunnerFromVM`, so that the implementation is symmetrical. Before you return the agentCommandRunner, you call `waitForRadyTimeout` if the option `WithWaitForReady` is true